src/clang-cpp-frontend/clang_cpp_convert.cpp

```diff
@@ -606,25 +606,14 @@ bool clang_cpp_convertert::get_expr(const clang::Stmt &stmt, exprt &new_expr)
     const clang::CXXThisExpr &this_expr =
       static_cast<const clang::CXXThisExpr &>(stmt);
 
-    std::size_t address =
-      reinterpret_cast<std::size_t>(current_functionDecl->getFirstDecl());
-
-    this_mapt::iterator it;
-    if(this_map.find(address) == this_map.end())
-    {
-      msg.error(fmt::format(
-        "Pointer `this' for method {} was not added to scope",
-        clang_c_convertert::get_decl_name(*current_functionDecl)));
-      abort();
-    }
-
     typet this_type;
     if(get_type(this_expr.getType(), this_type))
       return true;
 
-    assert(this_type == it->second.second);
+    std::string id, name;
+    get_decl_name(*current_functionDecl, name, id);
 
-    new_expr = symbol_exprt(it->second.first, it->second.second);
+    new_expr = symbol_exprt(id + "this", this_type);
```

Comment:
Would it be possible to use "current_thisID" to avoid implicit matching?

Reply:
(This shouldn't be part of this PR).

In theory, there shouldn't be any name conflict as the constructor name is unique. Even with multiple constructors, the first arg is always id + this.